<a href="https://colab.research.google.com/github/mariana-0/Proyecto_2/blob/master/opciones_proyectos.ipynb" target="_parent"><img src="https://colab.research.google.com/assets/colab-badge.svg" alt="Open In Colab"/></a>

# OPCIONES PROYECTO

In [1]:
import pandas as pd

## Opción I

In [5]:
filename1 = '/content/drive/MyDrive/CodingDojo_DataScience/P2/opcion healthcare-dataset-stroke-data.xlsx'
df1 = pd.read_excel(filename1)
df1.head()


id  gender   age  hypertension  heart_disease ever_married  \
0   9046    Male  67.0             0              1          Yes   
1  51676  Female  61.0             0              0          Yes   
2  31112    Male  80.0             0              1          Yes   
3  60182  Female  49.0             0              0          Yes   
4   1665  Female  79.0             1              0          Yes   

       work_type Residence_type  avg_glucose_level   bmi   smoking_status  \
0        Private          Urban             228.69  36.6  formerly smoked   
1  Self-employed          Rural             202.21   NaN     never smoked   
2        Private          Rural             105.92  32.5     never smoked   
3        Private          Urban             171.23  34.4           smokes   
4  Self-employed          Rural             174.12  24.0     never smoked   

   stroke  
0       1  
1       1  
2       1  
3       1  
4       1

In [8]:
df1.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 5110 entries, 0 to 5109
Data columns (total 12 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   id                 5110 non-null   int64  
 1   gender             5110 non-null   object 
 2   age                5110 non-null   float64
 3   hypertension       5110 non-null   int64  
 4   heart_disease      5110 non-null   int64  
 5   ever_married       5110 non-null   object 
 6   work_type          5110 non-null   object 
 7   Residence_type     5110 non-null   object 
 8   avg_glucose_level  5110 non-null   float64
 9   bmi                4909 non-null   float64
 10  smoking_status     5110 non-null   object 
 11  stroke             5110 non-null   int64  
dtypes: float64(3), int64(4), object(5)
memory usage: 479.2+ KB


### 1) Fuente de datos

Kagle: https://www.kaggle.com/datasets/fedesoriano/stroke-prediction-dataset

### 2) Breve descripción de datos

Conjunto de datos usados para predecir si un paciente es probable que tenga un ataque cerebral con base en características como genero, edad, enfermedades y si fuma o no.

### 3) ¿Cuál es el objetivo?

Predecir si tiene probabilidad de tener un ataque.

### 4) ¿Este es un problema de clasificación o regresión?

Clasificación

### 5) ¿Cuántas características hay?
12


### 6) ¿Cuántas filas de datos hay?
5110

### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?
Manejo de datos faltantes y datos 'desconocido' en la columna 'smoking_status'

## Opción II

In [6]:
filename2 = '/content/drive/MyDrive/CodingDojo_DataScience/P2/opcion Diamonds Prices2022.xlsx'
df2 = pd.read_excel(filename2)
df2.head()

count  carat      cut color clarity  depth  table  price     x     y     z
0      1   0.23    Ideal     E     SI2   61.5   55.0    326  3.95  3.98  2.43
1      2   0.21  Premium     E     SI1   59.8   61.0    326  3.89  3.84  2.31
2      3   0.23     Good     E     VS1   56.9   65.0    327  4.05  4.07  2.31
3      4   0.29  Premium     I     VS2   62.4   58.0    334  4.20  4.23  2.63
4      5   0.31     Good     J     SI2   63.3   58.0    335  4.34  4.35  2.75

In [9]:
df2.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 53943 entries, 0 to 53942
Data columns (total 11 columns):
 #   Column   Non-Null Count  Dtype  
---  ------   --------------  -----  
 0   count    53943 non-null  int64  
 1   carat    53943 non-null  float64
 2   cut      53943 non-null  object 
 3   color    53943 non-null  object 
 4   clarity  53943 non-null  object 
 5   depth    53943 non-null  float64
 6   table    53943 non-null  float64
 7   price    53943 non-null  int64  
 8   x        53943 non-null  float64
 9   y        53943 non-null  float64
 10  z        53943 non-null  float64
dtypes: float64(6), int64(2), object(3)
memory usage: 4.5+ MB


### 1) Fuente de datos
Kagle: https://www.kaggle.com/datasets/nancyalaswad90/diamonds-prices

### 2) Breve descripción de datos
Conjunto de datos que contienen precios y características de diamantes. La mayoría de las variables son numéricas, pero hay características que no lo son como corte, color y claridad.

### 3) ¿Cuál es el objetivo?
Predecir el precio de un diamante con base en algunas de sus características

### 4) ¿Este es un problema de clasificación o regresión?
Regresión

### 5) ¿Cuántas características hay?
10

### 6) ¿Cuántas filas de datos hay?

53943

### 7) ¿Qué retos, si es que hay alguno, prevén en la limpieza, exploración o modelización con este conjunto de datos?

Codificación de características no numéricas.